### PR TITLE
Azure fleet 2024-11-01 API updates.

### PR DESCRIPTION
**Pull request by rahuls-microsoft**
# Choose a PR Template

Switch to "Preview" on this description then select one of the choices below.

<a href="?expand=1&template=data_plane_template.md">Click here</a> to open a PR for a Data Plane API.

<a href="?expand=1&template=control_plane_template.md">Click here</a> to open a PR for a Control Plane (ARM) API.

### CODE REVIEW

File: specification/azurefleet/AzureFleet.Management/fleet.tsp [hunk 1]
```diff
@@ -211,6 +228,274 @@ union EvictionPolicy {
...
+  @extension("x-ms-identifiers", [])
+  @doc("Specifies which VMSizes should be excluded while building Fleet. Optional parameter.")
+  excludedVMSizes?: Array<string>;
+}
+
+@added(Versions.v2024_11_01)
+@doc("Represents the profile for a single additional location in the Fleet. The location and the virtualMachineProfileOverride (optional).")
+model LocationProfile {
+  @doc("""
+    The ARM location name of the additional region if LocationProfile is specified, then required.
```
Comment (SahilArora92) [inline specification/azurefleet/AzureFleet.Management/fleet.tsp:480]: Nit: Update the comment to this `The ARM location name of the additional region. If LocationProfile is specified, then location is required` or just `The ARM location name of the additional region.`

File: specification/azurefleet/AzureFleet.Management/examples/2023-11-01-preview/Fleets_CreateOrUpdate_MaximumSet.json [hunk 2]
```diff
@@ -765,7 +765,8 @@
     },
     "201": {
       "headers": {
-        "Azure-AsyncOperation": "https://contoso.com/operationstatus"
+        "Azure-AsyncOperation": "https://contoso.com/operationstatus",
+        "Retry-After": 30
```
Comment (SahilArora92) [inline specification/azurefleet/AzureFleet.Management/examples/2023-11-01-preview/Fleets_CreateOrUpdate_MaximumSet.json:769]: I don't see Retry-after being returned in response. Also retry after is optional field now from ARM. We use `x-ms-ratelimit-remaining-resource` for throttling. 
image

File: specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json [hunk 3]
```diff
@@ -0,0 +1,3581 @@
...
+          "$ref": "#/definitions/VMAttributeMinMaxDouble",
+          "description": "The range of memory specified from Min to Max. Must be specified if VMAttributes are specified, either Min or Max is required if specified."
+        },
+        "localStorageSupport": {
+          "$ref": "#/definitions/VMAttributeSupport",
+          "description": "Specifies whether the VMSize supporting local storage should be used to build Fleet or not."
+        },
+        "localStorageInGiB": {
+          "$ref": "#/definitions/VMAttributeMinMaxDouble",
+          "description": "The range of local storage in GB specified from Min to Max."
```
Comment (diyellap) [inline specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json:2470]: LocalStorageSupport should be set to \"Included\" or \"Required\" to use this VMAttribute. \nIf localStorageSupport is \"Excluded\", this VMAttribute can not be used." should be added to the description for localStorageinGiB.

File: specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json [hunk 4]
```diff
@@ -0,0 +1,3581 @@
...
+          "$ref": "#/definitions/VMAttributeMinMaxDouble",
+          "description": "The range of network bandwidth in Mbps specified from Min to Max. Optional parameter. Either Min or Max is required if specified."
+        },
+        "rdmaSupport": {
+          "$ref": "#/definitions/VMAttributeSupport",
+          "description": "Specifies whether the VMSize supporting RDMA (Remote Direct Memory Access) should be used to build Fleet or not."
+        },
+        "rdmaNetworkInterfaceCount": {
+          "$ref": "#/definitions/VMAttributeMinMaxInteger",
+          "description": "The range of RDMA (Remote Direct Memory Access) network interface count specified from Min to Max. Optional parameter. Either Min or Max is required if specified."
```
Comment (diyellap) [inline specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json:2498]: rdmaSupport should be set to \"Included\" or \"Required\" to use this VMAttribute. \nIf rdmaSupport is \"Excluded\", this VMAttribute can not be used." should be added to description for rdmaNetworkInterfaceCount.

File: specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json [hunk 5]
```diff
@@ -0,0 +1,3581 @@
...
+        },
+        "acceleratorSupport": {
+          "$ref": "#/definitions/VMAttributeSupport",
+          "description": "Specifies whether the VMSize supporting accelerator should be used to build Fleet or not."
+        },
+        "acceleratorManufacturers": {
+          "type": "array",
+          "description": "The accelerator manufacturers specified as a list. acceleratorSupport should be set to Included or Required to use this VMAttribute.",
```
Comment (diyellap) [inline specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json:2506]: acceleratorManufacturers should be set to "Included" or "Required" to use this VMAttribute. \nIf acceleratorManufacturers is "Excluded", this VMAttribute can not be used." should be added to description.

File: specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json [hunk 6]
```diff
@@ -0,0 +1,3581 @@
...
+          "items": {
+            "$ref": "#/definitions/AcceleratorManufacturer"
+          },
+          "x-ms-identifiers": []
+        },
+        "acceleratorTypes": {
+          "type": "array",
+          "description": "The accelerator types specified as a list. acceleratorSupport should be set to Included or Required to use this VMAttribute.",
```
Comment (diyellap) [inline specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json:2514]: acceleratorSupport should be set to "Included" or "Required" to use this VMAttribute. \nIf acceleratorSupport is "Excluded", this VMAttribute can not be used." should be added to description.

File: specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json [hunk 7]
```diff
@@ -0,0 +1,3581 @@
...
+          "items": {
+            "$ref": "#/definitions/AcceleratorType"
+          },
+          "x-ms-identifiers": []
+        },
+        "acceleratorCount": {
+          "$ref": "#/definitions/VMAttributeMinMaxInteger",
+          "description": "The range of accelerator count specified from min to max.. Optional parameter. Either Min or Max is required if specified."
```
Comment (diyellap) [inline specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json:2522]: acceleratorSupport should be set to "Included" or "Required" to use this VMAttribute. \nIf acceleratorSupport is "Excluded", this VMAttribute can not be used." should be added to description.

File: specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json [hunk 8]
```diff
@@ -0,0 +1,3581 @@
...
+          ]
+        }
+      }
+    },
+    "ArchitectureType": {
+      "type": "string",
+      "description": "Architecture types supported by Azure VMs.",
+      "enum": [
+        "ARM64",
+        "X86"
```
Comment (diyellap) [inline specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json:679]: This should be X64. Azure VMs are no longer supporting X86.

File: specification/azurefleet/AzureFleet.Management/examples/2023-11-01-preview/Fleets_CreateOrUpdate_MaximumSet.json (hunk 2 above)
Comment (rahuls-microsoft) [reply specification/azurefleet/AzureFleet.Management/examples/2023-11-01-preview/Fleets_CreateOrUpdate_MaximumSet.json]: Removed from Examples.

Comment (rahuls-microsoft) [reply specification/azurefleet/AzureFleet.Management/examples/2023-11-01-preview/Fleets_CreateOrUpdate_MaximumSet.json]: Removed it from examples.

File: specification/azurefleet/AzureFleet.Management/fleet.tsp (hunk 1 above)
Comment (rahuls-microsoft) [reply specification/azurefleet/AzureFleet.Management/fleet.tsp:480]: Updated.

File: specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json (hunk 3 above)
Comment (rahuls-microsoft) [reply specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json]: Updated.

File: specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json (hunk 4 above)
Comment (rahuls-microsoft) [reply specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json]: Updated.

File: specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json (hunk 5 above)
Comment (rahuls-microsoft) [reply specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json]: Updated.

File: specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json (hunk 6 above)
Comment (rahuls-microsoft) [reply specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json]: Updated.

File: specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json (hunk 7 above)
Comment (rahuls-microsoft) [reply specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json]: Updated.

File: specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json (hunk 8 above)
Comment (rahuls-microsoft) [reply specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json]: Updated.

**Comment by rahuls-microsoft** [comment]
@msyyc  - For Swagger BreakingChange  - There are few updates needed. The PR was merged couple of days back. No customer onboarded, so the changes should be fine. The supported CPU Architecture changed to X64 from X86.

File: specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json [hunk 9]
```diff
@@ -689,8 +689,8 @@
           },
           {
             "name": "X86",
```
Comment (chintanrp) [inline specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json:691]: Any specific reason we are king the name as "x86" for value "x64"? Should we not change the name as well to "x64"?

File: specification/azurefleet/AzureFleet.Management/fleet.tsp [hunk 10]
```diff
@@ -370,8 +370,8 @@ union ArchitectureType {
   @doc("ARM64 Architecture")
   ARM64: "ARM64",
 
-  @doc("X86 Architecture")
-  X86: "X86",
+  @doc("X64 Architecture")
+  X86: "X64",
```
Comment (chintanrp) [inline specification/azurefleet/AzureFleet.Management/fleet.tsp:374]: X64: "X64" ?

File: specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json (hunk 9 above)
Comment (diyellap) [reply specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json:691]: The name should be X64.

File: specification/azurefleet/AzureFleet.Management/fleet.tsp [hunk 11]
```diff
@@ -414,7 +414,10 @@ model VMAttributes {
   @doc("Specifies whether the VMSize supporting local storage should be used to build Fleet or not.")
   localStorageSupport?: VMAttributeSupport;
 
-  @doc("The range of local storage in GB specified from Min to Max.")
+  @doc("""
+    LocalStorageSupport should be set to "Included" or "Required" to use this VMAttribute. 
+    If localStorageSupport is "Excluded", this VMAttribute can not be used." should be added to the description for localStorageInGiB.
```
Comment (diyellap) [inline specification/azurefleet/AzureFleet.Management/fleet.tsp:419]: Can we please remove this? "**should be added to the description for localStorageInGiB**." from this?

File: specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json [hunk 12]
```diff
@@ -2467,7 +2470,7 @@
         },
         "localStorageInGiB": {
           "$ref": "#/definitions/VMAttributeMinMaxDouble",
-          "description": "The range of local storage in GB specified from Min to Max."
+          "description": "LocalStorageSupport should be set to \"Included\" or \"Required\" to use this VMAttribute. \nIf localStorageSupport is \"Excluded\", this VMAttribute can not be used.\" should be added to the description for localStorageInGiB."
```
Comment (diyellap) [inline specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json:2473]: Please remove "**should be added to the description for localStorageInGiB."** from description.

File: specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json (hunk 9 above)
Comment (rahuls-microsoft) [reply specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json:691]: updated.

File: specification/azurefleet/AzureFleet.Management/fleet.tsp (hunk 10 above)
Comment (rahuls-microsoft) [reply specification/azurefleet/AzureFleet.Management/fleet.tsp:374]: uupdated.

File: specification/azurefleet/AzureFleet.Management/fleet.tsp (hunk 11 above)
Comment (rahuls-microsoft) [reply specification/azurefleet/AzureFleet.Management/fleet.tsp:419]: updated.

File: specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json (hunk 12 above)
Comment (rahuls-microsoft) [reply specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json:2473]: Removed from tsp, the json will get updated.

File: specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json [hunk 13]
```diff
@@ -2495,31 +2498,31 @@
...
             "$ref": "#/definitions/AcceleratorType"
           },
           "x-ms-identifiers": []
         },
         "acceleratorCount": {
           "$ref": "#/definitions/VMAttributeMinMaxInteger",
-          "description": "The range of accelerator count specified from min to max.. Optional parameter. Either Min or Max is required if specified."
+          "description": "The range of accelerator count specified from min to max. Optional parameter. Either Min or Max is required if specified.\nacceleratorSupport should be set to \"Included\" or \"Required\" to use this VMAttribute. \nIf acceleratorSupport is \"Excluded\", this VMAttribute can not be used."
         },
         "vmCategories": {
```
Comment (diyellap) [inline specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json:2527]: Looks like we missed adding the following VMAttribute. This was a part of the API spec I shared. memoryInGiB and memoryInGiBPerVCpu are 2 different attributes. I see only memoryInGiB in this file.

/// <summary>
/// Optional.
/// The range of memory in GiB per vCPU specified from min to max.
/// Either min or max is required if specified.
/// </summary>
public VMAttributeMinMaxDouble memoryInGiBPerVCpu;

Comment (rahuls-microsoft) [reply specification/azurefleet/resource-manager/Microsoft.AzureFleet/stable/2024-11-01/azurefleet.json:2527]: Ok, I will add it, it was removed as per comment.

**Review by SahilArora92 — APPROVED** [review]

**Review by chintanrp — APPROVED** [review]

**Comment by rahuls-microsoft** [comment]
@tadelesh , @jsntcy , @qiaozha , @msyyc - Please check and sign off, so that we can proceed with SDK generation. Thanks for your help! we need to be ready for Iginite!

**Pull request merged**